crates/revm/src/optimism.rs

```diff
@@ -43,17 +43,29 @@ pub struct L1BlockInfo {
 }
 
 impl L1BlockInfo {
-    /// Fetches the L1 block info from the `L1Block` contract in the database.
-    pub fn try_fetch<DB: Database>(db: &mut DB) -> Result<L1BlockInfo, DB::Error> {
-        let l1_base_fee = db.storage(L1_BLOCK_CONTRACT, L1_BASE_FEE_SLOT)?;
-        let l1_fee_overhead = db.storage(L1_BLOCK_CONTRACT, L1_OVERHEAD_SLOT)?;
-        let l1_fee_scalar = db.storage(L1_BLOCK_CONTRACT, L1_SCALAR_SLOT)?;
-
-        Ok(L1BlockInfo {
-            l1_base_fee,
-            l1_fee_overhead,
-            l1_fee_scalar,
-        })
+    pub fn try_fetch<DB: Database>(
+        db: &mut DB,
+        is_optimism: bool,
+    ) -> Result<Option<L1BlockInfo>, EVMError<DB::Error>> {
```

Comment:
Are we sure we want to consider this an EVM error as opposed just bubbling up the `DB::Error`?

Slightly lean towards keeping it as a `DB::Error` since all this function is doing is trying to fetch the l1 block info from the database.

Reply:
Think you're right here - will update. Should probably be wrapped higher up if this fn is being used in the EVM.

Reply:
eg: https://github.com/anton-rs/op-revm/pull/20